### PR TITLE
Add extra step to check the health of istio version before update

### PR DESCRIPTION
**Pull request by jonathan34c**
edit: dev-infrastructure/scripts/istio.sh to check if the target vers…

Per discuss with AKS team, we need to check if the istio is up running before do the istio upgrade to makesure.

### What this PR does

Jira: 
Link to demo recording:

### CODE REVIEW

File: dev-infrastructure/scripts/istio.sh [hunk 1]
```diff
@@ -1,6 +1,13 @@
 #!/bin/bash
 
 set -euo pipefail
+ISTIO_NAMESPACE="aks-istio-system"
```
Comment (stevekuznetsov) [inline dev-infrastructure/scripts/istio.sh:4]: Can we use `az aks mesh versions`?

Comment (weinong) [reply dev-infrastructure/scripts/istio.sh:4]: use that for what?

Comment (stevekuznetsov) [reply dev-infrastructure/scripts/istio.sh:4]: For checking what versions of the `istio` control plane are available, without having to assume things like where the AKS plugin happens to run them, how they are annotated, whatever. Not sure we care about any of those implementation details, and we certainly do not care about making sure that we know how many replicas the plugin decides to run. What if the plugin thinks 5 replicas are necessary? Or 3? Or just 1?

Comment (jonathan34c) [reply dev-infrastructure/scripts/istio.sh:4]: the az aks mesh ```az aks mesh get-revisions --location``` only check the avilable versions to upgrade to in the certain location. it does not indicate the running istio in the current cluster

**Review by weinong — APPROVED** [review]
LGTM

**Review by mociarain — APPROVED** [review]

**Pull request merged**